src/lib/structures/Argument.ts

```diff
@@ -27,75 +26,112 @@ export enum ChannelType {
 	'GUILD_STAGE_VOICE' = 13,
 }
 
+export interface ArgumentChoice {
+	name: string;
+	value: string;
+}
+
 export interface ArgumentOptions {
 	name: string;
 	description: string;
 	type: ArgumentType | keyof typeof ArgumentType;
 	required?: boolean;
-	choices?: Array<CommandArgumentChoice>;
-	options?: Array<CommandArgument | Argument>;
+	choices?: Array<ArgumentChoice>;
+	arguments?: Array<Argument | ArgumentOptions>;
+	/**
+	 * @deprecated Please use ArgumentOptions#arguments instead
+	 * @link https://garlic-team.js.org/docs/#/docs/gcommands/next/typedef/ArgumentOptions
+	 */
+	options?: Array<Argument | ArgumentOptions>;
 	channelTypes?: Array<ChannelType | keyof typeof ChannelType>;
 	run?: (ctx: AutocompleteContext) => any;
 }
 
+const validationSchema = z
+	.object({
+		name: z
+			.string()
+			.max(32)
+			.regex(/^[a-zA-Z1-9]/),
+		description: z.string().max(100),
+		type: z
+			.union([z.string(), z.nativeEnum(ArgumentType)])
+			.transform((arg) =>
+				typeof arg === 'string' && Object.keys(ArgumentType).includes(arg) ? ArgumentType[arg] : arg,
+			),
+		required: z.boolean().optional(),
+		choices: z
+			.object({
+				name: z.string(),
+				value: z.string(),
+			})
+			.array()
+			.optional(),
+		arguments: z.any(),
+		channelTypes: z
+			.union([z.string(), z.nativeEnum(ChannelType)])
+			.transform((arg) => (Object.keys(ChannelType).includes(String(arg)) ? ChannelType[arg] : arg))
+			.array()
+			.optional(),
+		run: z.function().optional(),
+	})
+	.passthrough();
+
 export class Argument {
-	public readonly name: string;
-	public readonly description: string;
-	public readonly type: ArgumentType | keyof typeof ArgumentType;
-	public readonly required: boolean = false;
-	public readonly choices?: Array<CommandArgumentChoice>;
-	public readonly options?: Array<CommandArgument | Argument>;
-	public readonly channelTypes?: Array<ChannelType | keyof typeof ChannelType>;
+	public name: string;
+	public description: string;
+	public type: ArgumentType | keyof typeof ArgumentType;
+	public required = false;
+	public choices?: Array<ArgumentChoice>;
+	public arguments?: Array<Argument>;
+	public channelTypes?: Array<ChannelType | keyof typeof ChannelType>;
 	public run?: (ctx: AutocompleteContext) => any;
 
 	constructor(options: ArgumentOptions) {
-		Object.assign(this, options);
-
-		if (typeof this.type === 'string' && Object.keys(ArgumentType).includes(this.type))
-			this.type = ArgumentType[this.type];
+		if (options.options) {
+			Logger.warn('The use of ArgumentOptions#options is depracted. Please use ArgumentOptions#arguments instead');
```

Comment:
It's useless. It would be spamming. It's cool to just leave a comment and the IDE (Visual Studio Code, Webstorm) will cross out the function like it does with inhibitors.

Reply:
Maybe if I just remove the reassign? Some people might not know what a crossed-out line means, and as its deprecated, it will be removed in a future version. Just giving them this simple warning will help them solve, and migrate their commands, quick and easy. I do understand that it will cause some spam, but it would only spam until the user has resolved the depreciation issue.